### PR TITLE
SCRD-3808 Integrate barbican tests

### PR DESCRIPTION
**Pull request by mbelur**
This commit includes changes to run some of the
upstream service tests, more specifically, the barbican
client functional tests and the barbican functional API tests.
The tests run in a venv where the required packages are installed,
creates user/project resources, role assignment and the
configuration file needed  to run these tests.

### CODE REVIEW

File: scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/templates/tests/barbican-cli-func.sh.j2 [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+test_list_option=--list
+group_regex=([^\.]*\.)*
+
+EOF
+
+cp /var/lib/ardana/openstack/ardana/ansible/roles/tempest/files/tests2skip.py {{ ardana_qe_tests_dir }}/tests2skip.py
+
+http_proxy= https_proxy= HTTP_PROXY= HTTPS_PROXY= ${TESTR} run \
+    --subunit $($PYTHON ${TESTS_TO_SKIP_BIN} ${RUN_FILTER})|
+    ${VENV_FOLDER}/bin/subunit-trace --no-failure-debug -f | tee testr_results.log
```
Comment (flaviodsr) [inline scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/templates/tests/barbican-cli-func.sh.j2]: Please save the test output to `{{ ardana_qe_test_log }}` so jenkins can store the test log.

File: scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/templates/tests/barbican-functional.sh.j2 [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+test_list_option=--list
+group_regex=([^\.]*\.)*
+
+EOF
+
+cp /var/lib/ardana/openstack/ardana/ansible/roles/tempest/files/tests2skip.py {{ ardana_qe_tests_dir }}/tests2skip.py
+
+http_proxy= https_proxy= HTTP_PROXY= HTTPS_PROXY= ${TESTR} run \
+    --subunit $($PYTHON ${TESTS_TO_SKIP_BIN} ${RUN_FILTER})|
+    ${VENV_FOLDER}/bin/subunit-trace --no-failure-debug -f | tee testr_results.log
```
Comment (flaviodsr) [inline scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/templates/tests/barbican-functional.sh.j2]: Please save the test output to `{{ ardana_qe_test_log }}` so jenkins can store the test log.

File: scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/vars/barbican-functional.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  skipped: "awk '/- Skipped:/ { print $3 }' {{ ardana_qe_test_log }}"
+
+ardana_qe_test_venv_requires:
+  - 'os-testr'
+ardana_qe_test_run_filters: "{{ role_path }}/files/run_filters/{{ test_name }}.txt"
+os_resources_requires:
+  - 'user'
+  - 'project'
+
+os_resource_test_auth:
```
[the remark this answers is not in the document]
Comment (mbelur) [reply scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/vars/barbican-functional.yml]: Updated in the latest commit

File: scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/defaults/main.yml [hunk 4]
```diff
@@ -43,7 +45,4 @@ os_resource_state: present
 os_resource_prefix: "ansible-qe-tests-{{ test_name }}"
 os_resources_requires: []
 os_resource_test_auth:
-  username: "{{ os_resource_prefix }}-user"
-  password: "secret"
-  project: "{{ os_resource_prefix }}-project"
-  role: "Member"
+  - {username: "{{ os_resource_prefix }}-user",password: "secret",project: "{{ os_resource_prefix }}-project",role: "Member"}
```
[the remark this answers is not in the document]
Comment (mbelur) [reply scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/defaults/main.yml]: done in latest commit

File: scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/tasks/main.yml [hunk 5]
```diff
@@ -15,6 +15,8 @@
 #
 ---
 
+- include_tasks: read_osrc.yml
```
[the remark this answers is not in the document]
Comment (mbelur) [reply scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/tasks/main.yml]: The reason being that I refer to the os_admin_osrc in the template scripts - barbican-cli-func.sh.j2 and barbican-funtional.sh.j2 and when the prepare_test_env.yml playbook runs which copies these test run scripts, it fails with os_admin_osrc var not defined. So I had to seperate it out and call this before prepare_test_evn.yml.
Also by seperating this out, we can extend the read_osrc.yml playbook to support reading the deployer group_vars for any other additional environment propertiees.

File: scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/tasks/prepare_test_env.yml [hunk 6]
```diff
@@ -42,9 +43,17 @@
     dest: "{{ ardana_qe_test_work_dir }}/{{ test_name }}.sh"
     mode: "0755"
 
+- name: create run_filters directory
```
[the remark this answers is not in the document]
Comment (mbelur) [reply scripts/jenkins/ardana/ansible/roles/ardana_qe_tests/tasks/prepare_test_env.yml]: done in latest commit

**Comment by flaviodsr** [comment]
What's the purpose of `run_filter`, seems like it has the same value as `test_name`?

I suggest setting the default value of `test_branch` to `stable/pike` if cloud8 and `stable/rocky` if cloud9 and removing the `run_filter` parameter (or at least set a default value for it). The idea is that **anyone** should be able to run the tests, at least with default values, without requiring previous knowledge about it (such as which `test_filter`/`test_branch` I should use for an specific test).

**Comment by mbelur** [comment]
I will remove having test_name for the run_filter. I would like to keep this parameter since run_filters can give extra filtering of tests if needed. For example: if I want to run only the smoke tests for barbican-functional, then I can define the smoke run_filter for it. The run_filter parameter exists earlier in the framework.
Regarding test_branch value, I can temporarily patch the barbican template scripts to use the right branch based on the release version of the cloud. But however, this should be part of the ansible playbook support from framework point of view.

**Comment by mbelur** [comment]
@flaviodsr I have squashed all my commits into one single commit as requested. Please review.

**Pull request merged**